lib/cloud-metadata/aws.js

```diff
@@ -1,25 +1,29 @@
 'use strict'
+const URL = require('url').URL
 const { httpRequest } = require('../http-request')
-const { cleanupRequest } = require('./shared')
-function getLogger () {
-  return require('../..').logger
+const DEFAULT_BASE_URL = new URL('/', 'http://169.254.169.254:80')
```

Comment:
nit: Any particular reason to not use `new URL('http://169.254.169.254/')` ?

```
> DEFAULT_BASE_URL = new URL('/', 'http://169.254.169.254:80')
URL {
  href: 'http://169.254.169.254/',
  origin: 'http://169.254.169.254',
  protocol: 'http:',
  username: '',
  password: '',
  host: '169.254.169.254',
  hostname: '169.254.169.254',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
> DEFAULT_BASE_URL.toString()
'http://169.254.169.254/'
```

Reply:
> Any particular reason to not use
A desire to be consistent with how we needed to create URLs in our tests (with a port that's not `80`), an unconscious dislike for APIs where parameters have different meanings in variadic context, and an uncertainty around how the `URL` object handles things across node versions when a port's omitted. 

It's a small change that I'm happy to make. Also what's the general rule we're following here -- this is the sort of thing I'd prefer a linter warn us about.

Reply:
"nit:", so no biggie either way.   I found using the second `base` argument to `URL(...)` to be a little odd is all.

https://nodejs.org/api/url.html#url_class_url is stable since node v7.

> this is the sort of thing I'd prefer a linter warn us about.

Yah, I don't have a great general suggestion. I don't have experience with custom eslint rules/recipes. There is a deep end we could fall off going down that path as well.